src/cpu/pred/btb/microtage.cc

```
@@ -0,0 +1,1098 @@
+        //initialize ittage predictor
+        assert(tableSizes.size() >= numPredictors);
+        tageTable[i].resize(tableSizes[i]);
+        for (unsigned int j = 0; j < tableSizes[i]; ++j) {
+            tageTable[i][j].resize(numWays);
+        }
+
+        tableIndexBits[i] = ceilLog2(tableSizes[i]);
+        tableIndexMasks[i].resize(tableIndexBits[i], true);
+
+        assert(histLengths.size() >= numPredictors);
+
```

Cppcheck (2.19.0)

[error] Reference to temporary returned.

(returnTempReference)